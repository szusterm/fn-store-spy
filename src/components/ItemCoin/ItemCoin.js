import React, {Component} from 'react';
import PropTypes from 'prop-types';

export class ItemCoin extends Component {
	render() {
		const {imageSrc, rarity} = this.props;

		return (
			<div className={`item-coin item-coin--${rarity}`}>
				<img
					className={'item-coin--image'}
					src={imageSrc}
				/>
			</div>
		);
	}
}

ItemCoin.propTypes = {
	_id: PropTypes.string.isRequired,
	imageSrc: PropTypes.string.isRequired,
	rarity: PropTypes.string.isRequired
};

export default ItemCoin;