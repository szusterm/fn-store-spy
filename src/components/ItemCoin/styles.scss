@import '../../scss/rarities-colors';
@import '../../scss/shadow';

.item-coin {
	width: 100%;
	height: auto;

	border-radius: 50%;
	border: 3px solid;

	@include shadow(sm);

	&--uncommon {
		background: $gradient-uncommon;
		border-color: $border-uncommon;
	}
	&--rare {
		background: $gradient-rare;
		border-color: $border-rare;
	}
	&--epic {
		background: $gradient-epic;
		border-color: $border-epic;
	}
	&--legendary {
		background: $gradient-legendary;
		border-color: $border-legendary;
	}

	.item-coin {
		&--image {
			width: 100%;
			height: 100%;

			border-radius: 50%;
		}
	}
}