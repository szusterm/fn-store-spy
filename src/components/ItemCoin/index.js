import ItemCoin from './ItemCoin';

export default ItemCoin;