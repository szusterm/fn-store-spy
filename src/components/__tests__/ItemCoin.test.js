import React from 'react';
import {shallow} from 'enzyme';

import {ItemCoin} from '../ItemCoin/ItemCoin';

const setup = (propOverrides) => {
	const props = Object.assign({
		imageSrc: 'https://example.com',
		rarity: 'uncommon'
	}, propOverrides);

	const wrapper = shallow(<ItemCoin {...props}/>);

	return {props, wrapper};
};

describe('CircleOrderCount Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('adds class with background rarity color', () => {
		const {wrapper, props} = setup({
			rarity: 'uncommon'
		});

		const rarityColorClass = `.item-coin--${props.rarity}`;
		const displayedItem = wrapper.find(rarityColorClass);

		expect(displayedItem.exists()).toBe(true);
	});

	it('shows an item image from prop', () => {
		const {wrapper, props} = setup({
			imageSrc: 'https://example.com'
		});

		const displayedImgSrc = wrapper.find('.item-coin--image').props().src;

		expect(displayedImgSrc).toBe(props.imageSrc);
	});
});