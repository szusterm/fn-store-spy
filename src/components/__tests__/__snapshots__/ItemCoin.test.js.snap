// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleOrderCount Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemCoin
    imageSrc="https://example.com"
    rarity="uncommon"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        className="item-coin--image"
        src="https://example.com"
      />,
      "className": "item-coin item-coin--uncommon",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "item-coin--image",
        "src": "https://example.com",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          className="item-coin--image"
          src="https://example.com"
        />,
        "className": "item-coin item-coin--uncommon",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "item-coin--image",
          "src": "https://example.com",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
